pkg/image/fetcher.go

```diff
@@ -171,8 +178,29 @@ func (f *Fetcher) fetchDaemonImage(name string) (imgutil.Image, error) {
 	return image, nil
 }
 
-func (f *Fetcher) fetchRemoteImage(name string) (imgutil.Image, error) {
-	image, err := remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
+func (f *Fetcher) fetchRemoteImage(name string, target *dist.Target) (imgutil.Image, error) {
+	var (
+		image imgutil.Image
+		err   error
+	)
+
+	if target == nil {
+		image, err = remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
+	} else {
+		platform := imgutil.Platform{OS: target.OS, Architecture: target.Arch}
+		// when targeting a registry, we need to use variant and osVersion to hit the correct image
```

Comment:
```suggestion
		// when targeting a registry, we need to use variant if available to hit the correct image
```
I think this is more accurate?